engine/src/witness/btc/btc_deposits.rs

```diff
@@ -28,6 +28,7 @@ impl<Inner: ChunkedByVault> ChunkedByVaultBuilder<Inner> {
 	pub fn btc_deposits<ProcessCall, ProcessingFut>(
 		self,
 		process_call: ProcessCall,
+		is_prewitness: bool,
```

Comment:
Why cannot whatever the logic instead here just go into the process_call?

Reply:
Good point, this is a rememnant of a previous approach, we should do it this way

Reply:
It is difficult to do while keeping the `ProcessCall` generic. Maybe i'm not understanding. Can you dm me with details if you have time.